PAWS/PAWSStandard.swift

```diff
@@ -81,25 +81,39 @@ actor PAWSStandard: Standard, EnvironmentAccessible, HealthKitConstraint, Onboar
     
     private func upload(electrocardiogram: HKElectrocardiogram) async {
         var supplementalMetrics: [HKSample] = []
+        var precedingPulseRates: [HKQuantitySample] = []
+        var precedingPhysicalEffort: [HKQuantitySample] = []
+        var precedingStepCount: [HKQuantitySample] = []
+        var precedingActiveEnergy: [HKQuantitySample] = []
+        var precedingVo2Max: HKQuantitySample?
         
         do {
             try await upload(sample: electrocardiogram)
-            
-            supplementalMetrics.append(contentsOf: try await electrocardiogram.precedingPulseRates)
-            supplementalMetrics.append(contentsOf: try await electrocardiogram.precedingPhysicalEffort)
-            supplementalMetrics.append(contentsOf: try await electrocardiogram.precedingStepCount)
-            supplementalMetrics.append(contentsOf: try await electrocardiogram.precedingActiveEnergy)
-            
-            if let precedingVo2Max = try await electrocardiogram.precedingVo2Max {
-                supplementalMetrics.append(precedingVo2Max)
-            }
-            
-            for supplementalMetric in supplementalMetrics {
-                try await upload(sample: supplementalMetric)
-            }
+            precedingPulseRates = try await electrocardiogram.precedingPulseRates
+            precedingPhysicalEffort = try await electrocardiogram.precedingPhysicalEffort
+            precedingStepCount = try await electrocardiogram.precedingStepCount
+            precedingActiveEnergy = try await electrocardiogram.precedingActiveEnergy
+            precedingVo2Max = try await electrocardiogram.precedingVo2Max
```

Comment:
Couldn't we just use a `try?` statement here and provide an empty collection as a default value. Even with this change, of the `electrocardiogram.precedingPulseRates` call fails none of the other ones will be uploaded.

So closer to the old version but with some `try?` calls?

Reply:
Yep, that makes sense. I'll incorporate those changes right away.